lib/matplotlib/widgets.py

```diff
@@ -856,23 +856,30 @@ def _notify_change_observers(self):
 
     def begin_typing(self, x):
         self.capturekeystrokes = True
-        # disable command keys so that the user can type without
-        # command keys causing figure to be saved, etc
-        self.reset_params = {}
-        for key in self.params_to_disable:
-            self.reset_params[key] = rcParams[key]
-            rcParams[key] = []
+        if rcParams['toolbar'] != 'toolmanager':
```

Comment:
Should rather check whether the figure manager's key_press_handler_id is not None, otherwise this check becomes wrong if the value of the rcParam is changed between the time the canvas is created and the time begin_typing is called.

Same below.